py/fiberassign/fba_launch_io.py

```diff
@@ -1388,6 +1386,42 @@ def create_mtl(
                 targdir, tiles=tiles, quick=True, columns=columns + ["TARGETID"]
             )
             d = match_ledger_to_targets(d, targ)
+    # AR mtl: case secondary
+    else:
+        # AR mtl: secondary for backup should never happen, but safe approach still
+        if ("backup" not in mtldir) & (desitarget.__version__ >= "1.2.2"):
```

Comment:
I assume the version check is for backward compatibility?  That makes sense.  This looks like a plain alphabetical string comparison test, though, which I don't think is what you want.  You might want something like:
https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python
though I don't think that will handle cases like __version__ == 'master' or something.  Is there a test we can do on a different string to get the behavior you want?

Reply:
thanks for noticing that.

I have used that kind of pure-string comparison for fiberassign and desitarget versions in lots of place in fba_launch_io.py, fba_rerun_io.py, and fba_rerun...
that is working from the tests I have done (even with master); but I acknowledge it will break if we reach one day versions 10.0.0 (hopefully that shouldn t happen!).

I could use this packaging.version.parse() with a test for the "master" case; I can investigate if there are other solutions.
pinging @sbailey in case he is aware of other options.